Added an example to IIS rewrites article (#6711)
umbraco/UmbracoDocs: 15/umbraco-cms/reference/routing/iisrewriterules.md

## Examples of rewrite rules

Below are some handy IIS Rewrite Rules you can use in your projects:

* **10 Handy IIS Rewrite Rules**

A great resource showcasing 10 practical IIS Rewrite rules: [URL rewriting tips and tricks](https://ruslany.net/2009/04/10-url-rewriting-tips-and-tricks/)

* **Examples of IIS Rewrite Rules**

Another useful collection of IIS rewrite rule examples:  [Some useful IIS rewrite rules](https://odetocode.com/blogs/scott/archive/2014/03/27/some-useful-iis-rewrite-rules.aspx)

> * **Static Rewrites Using Rewrite Maps**

Reviewer: I don't really understand why this is changed? 🤔 
It looks a bit strange in the preview:

![image](https://github.com/user-attachments/assets/cb652755-6ef0-463c-85b9-2b1311a073a4)

Author: It's interesting how it doesn't look like the way I expected it. 

I noticed that the article starts with headings followed by text, and then shifts to just text with code samples at the bottom. To maintain the flow I changed it this way. However, if you prefer, I can leave it as it is and just add headings to the code snippets below.

Reviewer: I think the list looks better at least.
I do see your point about the consistency.. Would it make sense to give the list a little sub heading as well - something like "External examples" or "External resources"? 🤔 
And then a little sentence or two, about what the links are for? 🤔

Author: I've added a sub-heading and a 1 liner description to it. Let me know what you think.

If you need to handle a lot of static rewrites, consider using rewrite maps: [Rule with rewrite map rule template](https://www.iis.net/learn/extensions/url-rewrite-module/rule-with-rewrite-map-rule-template)

### Example: Remove a Trailing Slash

> To remove a trailing slash from the URL (ensure Umbraco does not add a trailing slash by setting `AddTrailingSlash` to `false` in your [RequestHandler settings](../configuration/requesthandlersettings.md)):

Reviewer: Might make sense to move the second sentence below the code snippet, but I'll leave that up to you to decide!

```suggestion
The following rule removes any trailing slashes from the URL.

Ensure Umbraco does not add a trailing slash by setting `AddTrailingSlash` to `false` in your [RequestHandler settings](../configuration/requesthandlersettings.md).
```

</rule>
```

### Example: Enforce HTTPS

> To ensure your site only runs on HTTPS:

Reviewer: ```suggestion
The following rule ensures your site only runs on HTTPS:
```

</rule>
```

### Example: Redirect Non-www to www

> To redirect traffic from non-www to www (excluding the Umbraco Cloud project hostname):

Reviewer: Should the part about Cloud be added in a note? Otherwise it might be easy to miss 🤔 

```suggestion
The following rule redirects traffic from non-www to www (excluding the Umbraco Cloud project hostname):
```

</rule>
```

### Example: Remove the .aspx Extension

> If you want to redirect .aspx URLs to their extensionless counterparts, you can use the following rule. Make sure you also have a web.config file in the root of your site:

Reviewer: I'm not sure the sentence about the `web.config` is necessary, as all the examples above also need to go into the `web.config` file, right? 🤔 

```suggestion
The following rule redirects `.aspx` URLs to their extensionless counterparts. Make sure you also have a `web.config` file in the root of your site.
```

Author: Yes, you are right. I missed it was mentioned in the Note above.

          <match url="^(.*)\.aspx$" />
          <conditions logicalGrouping="MatchAll">
            <add input="{REQUEST_FILENAME}" matchType="IsFile" negate="true" />
            <add input="{REQUEST_FILENAME}" matchType="IsDirectory" negate="true" />
          </conditions>
          <action type="Redirect" url="{R:1}" redirectType="Permanent" />
        </rule>
      </rules>
    </rewrite>
  </system.webServer>
</configuration>
```

### Example: Custom Rewrite Rules for Umbraco Cloud

An example configuration to help ensure your custom rules integrate properly:

```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration xmlns:xdt="http://schemas.microsoft.com/XML-Document-Transform">
  <location path="." inheritInChildApplications="false">
    <system.webServer>
      <rewrite xdt:Transform="Insert">
        <rules>
          <!-- Add your custom rules here -->
        </rules>
      </rewrite>
    </system.webServer>
  </location>
</configuration>
```

{% hint style="info" %}
If you use **Umbraco Cloud**, check the [Rewrite Rules](https://docs.umbraco.com/umbraco-cloud/set-up/project-settings/manage-hostnames/rewrites-on-cloud) article.

> {% endhint %}

Reviewer: I'm not aware whether this is already in the Cloud docs, but in any case, I think it would make more sense over there? 🤔

Author: I will add it to the Cloud docs as well. I added it here too cz all the examples mentioned in the article can be used for both CMS and cloud.
